pkgs/applications/blockchains/lighthouse/default.nix

```
@@ -0,0 +1,102 @@
+{ clang
+, cmake
+, darwin
+, fetchFromGitHub
+, fetchurl
+, lib
+, lighthouse
+, llvmPackages
+, nodePackages
+, perl
+, protobuf
+, runCommand
+, rustPlatform
+, stdenv
+, testers
+, unzip }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "lighthouse";
+  version = "3.1.0";
+
+  # lighthouse/common/deposit_contract/build.rs
+  depositContractSpecVersion = "0.12.1";
+  testnetDepositContractSpecVersion = "0.9.2.1";
+
+  __darwinAllowLocalNetworking = true;
+
+  src = fetchFromGitHub {
+    owner = "sigp";
+    repo = "lighthouse";
+    rev = "v${version}";
+    hash = "sha256-HaoqpwCWyP9A4qWfwDDED4pSsk82oI0ZiipkNJLcIBY=";
+  };
+
+  cargoHash = "sha256-6+jOHRnOyj0aVS1v/7U72wP5ZWcuG2J9RQYCbgDgZFM=";
+
+  buildFeatures = [ "modern" "gnosis" ];
+
+  checkFeatures = [ ];
+
+  nativeBuildInputs = [ clang cmake perl protobuf ];
+
+  buildInputs = lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Security
+  ];
+
+  LIBCLANG_PATH = "${llvmPackages.libclang.lib}/lib";
+
+  depositContractSpec = fetchurl {
+    url = "https://raw.githubusercontent.com/ethereum/eth2.0-specs/v${depositContractSpecVersion}/deposit_contract/contracts/validator_registration.json";
+    hash = "sha256-ZslAe1wkmkg8Tua/AmmEfBmjqMVcGIiYHwi+WssEwa8=";
+  };
+
+  testnetDepositContractSpec = fetchurl {
+    url = "https://raw.githubusercontent.com/sigp/unsafe-eth2-deposit-contract/v${testnetDepositContractSpecVersion}/unsafe_validator_registration.json";
+    hash = "sha256-aeTeHRT3QtxBRSNMCITIWmx89vGtox2OzSff8vZ+RYY=";
+  };
+
+  LIGHTHOUSE_DEPOSIT_CONTRACT_SPEC_URL = "file://${depositContractSpec}";
+  LIGHTHOUSE_DEPOSIT_CONTRACT_TESTNET_URL = "file://${testnetDepositContractSpec}";
+
+  cargoBuildFlags = [
+    "--package" "lighthouse"
+  ];
+
+  # All of these tests require network access
+  cargoTestFlags = [
+    "--workspace"
+    "--exclude" "beacon_node"
+    "--exclude" "http_api"
+    "--exclude" "beacon_chain"
+    "--exclude" "lighthouse"
+    "--exclude" "lighthouse_network"
+    "--exclude" "slashing_protection"
+    "--exclude" "web3signer_tests"
+  ];
+
+  checkFlags = [
+    "--skip" "service::tests::tests::test_dht_persistence"
+    "--skip" "time::test::test_reinsertion_updates_timeout"
+  ] ++ lib.optionals (stdenv.isAarch64 && stdenv.isDarwin) [
+    "--skip" "subnet_service::tests::sync_committee_service::same_subscription_with_lower_until_epoch"
+    "--skip" "subnet_service::tests::sync_committee_service::subscribe_and_unsubscribe"
+  ];
+
+  checkInputs = [
+    nodePackages.ganache
+  ];
+
+  passthru.tests.version = testers.testVersion {
+    package = lighthouse;
+    command = "lighthouse --version";
+    version = "v${lighthouse.version}";
+  };
+
+  meta = with lib; {
+    description = "Ethereum consensus client in Rust";
+    homepage = "https://lighthouse.sigmaprime.io/";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ centromere ];
```

Since you are building with Darwin in mind.
```suggestion
    maintainers = with maintainers; [ centromere ];
    platforms = platforms.unix;
```

---

Isn't `platforms.unix ++ platforms.darwin` the default?

---

Yep, looks like it [is](https://github.com/NixOS/nixpkgs/blob/2da64a81275b68fdad38af669afeda43d401e94b/pkgs/build-support/rust/build-rust-package/default.nix#L157).